wsdl-core/src/main/java/io/ballerina/wsdl/core/WsdlToBallerina.java

```diff
@@ -18,138 +18,392 @@
+    public static final String QUOTATION_MARK = "\"";
+    public static final String XMLDATA_PARSE_AS_TYPE = "xmldata:parseAsType";
+    public static final String SOAP11_NAMESPACE = "http://schemas.xmlsoap.org/soap/envelope/";
+    public static final String SOAP12_NAMESPACE = "http://www.w3.org/2003/05/soap-envelope";
+    public static final String HEADER = "Header";
+    public static final String SLASH = "/";
+    public static final String XMLDATA_NAMESPACE_URI = "@xmldata:Namespace {uri: \"%s\"}\n%s %s?;";
+    public static final String MISSING_HEADER_ELEMENT_ERROR = "Header element name cannot be extracted.";
+    public static final String MISSING_DATA_IN_HEADER_ELEMENT_ERROR = "Header element is not found in the WSDL Definition: ";
+    public static final String OPERATION_NOT_FOUND_ERROR = "WSDL operation is not found: ";
     private Definition wsdlDefinition;
     private Port soapPort;
     private SoapVersion soapVersion;
+    private String soapNamespace;
+    private String serviceUrl;
 
-    public WsdlToBallerinaResponse generateFromWSDL(String wsdlDefinitionText, List<String> filteredWSDLOperations) {
-        List<DiagnosticMessage> diagnosticMessages = new ArrayList<>();
-        WsdlToBallerinaResponse response = new WsdlToBallerinaResponse();
+    public void generateFromWSDL(WsdlToBallerinaResponse response, Definition wsdlDefinition,
+                                 String outputDirectory, List<DiagnosticMessage> diagnosticMessages,
+                                 String[] filteredWSDLOperations) {
         try {
-            WSDLReader reader = WSDLFactory.newInstance().newWSDLReader();
-            reader.setFeature("javax.wsdl.verbose", false);
-            reader.setFeature("javax.wsdl.importDocuments", true);
-            InputStream wsdlStream = new ByteArrayInputStream(wsdlDefinitionText.getBytes(Charset.defaultCharset()));
-            wsdlDefinition = reader.readWSDL(null, new InputSource(wsdlStream));
-
+            this.wsdlDefinition = wsdlDefinition;
             WsdlService wsdlService = getSoapService(wsdlDefinition);
             if (wsdlService == null) {
-                return response;
+                DiagnosticMessage message = DiagnosticMessage.wsdlToBallerinaError(null);
+                diagnosticMessages.add(message);
+                DiagnosticUtils.getDiagnosticResponse(diagnosticMessages, response);
+                return;
             }
-
             soapVersion = wsdlService.getSoapVersion();
+            serviceUrl = wsdlService.getSoapServiceUrl();
+            soapNamespace = soapVersion.equals(SoapVersion.SOAP12) ? SOAP12_NAMESPACE : SOAP11_NAMESPACE;
             initializeSchemas(wsdlDefinition);
+            Map<String, WsdlOperation> wsdlOperations = getWSDLOperations();
+            Types types = wsdlDefinition.getTypes();
+            if (types == null) {
+                throw new Exception("Could not find <wsdl:types> in the file");
+            }
```

Comment:
I think we can use `Objects.requireNotNull` [1] here

[1] - https://docs.oracle.com/javase/8/docs/api/java/util/Objects.html#requireNonNull-T-java.lang.String-

Reply:
Fixed in [9b99cff](https://github.com/ballerina-platform/wsdl-tools/pull/17/commits/9b99cff913e103560cca297a4f8abb9ee7ae383c)